### PR TITLE
Add new LXC: Vikunja

### PR DESCRIPTION
**Pull request by MickLesk**
## Description
@judeibe @newzealandpaul @Mellowlynx 

You can test it here: 
bash -c "$(wget -qLO - https://raw.githubusercontent.com/MickLesk/Proxmox_DEV/main/ct/vikunja.sh)"

Please give Feedback to merge the code. 

## Type of change
Please check the relevant option(s):
- [x] New script (a fully functional and thoroughly tested script or set of scripts.)
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Documentation update required (this change requires an update to the documentation)

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
https://github.com/community-scripts/ProxmoxVE/discussions/35

@BramSuurdje for website:
Description
Vikunja is a powerful self-hosted todo app.  It allows you to create and manage to-do lists. You can plan tasks, set priorities and collaborate with others. The best part is that your data is safe with you and you can customize the app to your liking. It's like a personal assistant that helps you stay organized.

How to install
To create a new Vikunja LXC run the command below in the Proxmox VE Shell.

bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/vikunja.sh)"

### CODE REVIEW

File: ct/vikunja.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+function update_script() {
+header_info
+if [[ ! -d /opt/vikunja ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+if (( $(df /boot | awk 'NR==2{gsub("%","",$5); print $5}') > 80 )); then
+  read -r -p "Warning: Storage is dangerously low, continue anyway? <y/N> " prompt
+  [[ ${prompt,,} =~ ^(y|yes)$ ]] || exit
+fi
+RELEASE=$(curl -s https://dl.vikunja.io/vikunja/ | grep -oP 'href="/vikunja/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
+if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+  msg_info "Stopping Services"
```
Comment (havardthom) [inline ct/vikunja.sh:66]:
```suggestion
  msg_info "Stopping ${APP}"
```

File: ct/vikunja.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  systemctl stop vikunja
+  msg_ok "Services Stopped"
```
Comment (havardthom) [inline ct/vikunja.sh:68]:
```suggestion
  msg_ok "Stopped ${APP}"
```

File: ct/vikunja.sh [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  msg_ok "Removed Old Installation"
+
+  msg_info "Updating ${APP} to ${RELEASE}"
+  cd /opt
+  wget -q "https://dl.vikunja.io/vikunja/$RELEASE/vikunja-$RELEASE-amd64.deb"
+  DEBIAN_FRONTEND=noninteractive dpkg -i vikunja-$RELEASE-amd64.deb &>/dev/null
+  echo "${RELEASE}" >/opt/${APPLICATION}_version.txt
+  msg_ok "Updated ${APP}"
+
+  msg_info "Starting Services"
```
Comment (havardthom) [inline ct/vikunja.sh:81]:
```suggestion
  msg_info "Starting ${APP}"
```

File: ct/vikunja.sh [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  systemctl start vikunja
+  msg_ok "Started Services"
```
Comment (havardthom) [inline ct/vikunja.sh:83]:
```suggestion
  msg_ok "Started ${APP}"
```

File: ct/vikunja.sh [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  read -r -p "Warning: Storage is dangerously low, continue anyway? <y/N> " prompt
+  [[ ${prompt,,} =~ ^(y|yes)$ ]] || exit
+fi
+RELEASE=$(curl -s https://dl.vikunja.io/vikunja/ | grep -oP 'href="/vikunja/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
+if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+  msg_info "Stopping Services"
+  systemctl stop vikunja
+  msg_ok "Services Stopped"
+
+  msg_info "Remove Old Installation"
```
Comment (havardthom) [inline ct/vikunja.sh:70]:
```suggestion
```

File: ct/vikunja.sh [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  rm -rf /opt/*
+  msg_ok "Removed Old Installation"
```
Comment (havardthom) [inline ct/vikunja.sh:72]:
```suggestion
```

File: install/vikunja-install.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2021-2024 communtiy-scripts ORG
```
Comment (havardthom) [inline install/vikunja-install.sh:3]:
```suggestion
# Copyright (c) 2021-2024 community-scripts ORG
```
also wrong in `bookstack-install.sh`

File: install/vikunja-install.sh [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+source /dev/stdin <<< "$FUNCTIONS_FILE_PATH"
+color
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies (Patience)"
```
Comment (havardthom) [inline install/vikunja-install.sh:16]:
```suggestion
msg_info "Installing Dependencies"
```

File: install/vikunja-install.sh [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+  make \
+  mc
+msg_ok "Installed Dependencies"
+
+msg_info "Setup Vikunja (Patience)"
+cd /opt
+RELEASE=$(curl -s https://dl.vikunja.io/vikunja/ | grep -oP 'href="/vikunja/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
+wget -q "https://dl.vikunja.io/vikunja/$RELEASE/vikunja-$RELEASE-amd64.deb"
+$STD dpkg -i vikunja-$RELEASE-amd64.deb
+systemctl start vikunja.service
```
Comment (havardthom) [inline install/vikunja-install.sh:29]:
```suggestion
```
think `dpkg` already enables and starts service

File: ct/vikunja.sh [hunk 10]
```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
+# Copyright (c) 2021-2024 co
```
Comment (havardthom) [inline ct/vikunja.sh:3]:
```suggestion
# Copyright (c) 2021-2024 community-scripts ORG
```

File: ct/vikunja.sh [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+
+  msg_info "Cleaning Up"
+  rm -rf /opt/vikunja-$RELEASE-amd64.deb
+  msg_ok "Cleaned"
+  msg_ok "Updated Successfully"
+else
+  msg_ok "No update required. ${APP} is already at ${RELEASE}"
+fi
+exit
+}
```
Comment (havardthom) [inline ct/vikunja.sh:93]:
```suggestion
}

```

File: CHANGELOG.md [hunk 12]
```diff
@@ -19,6 +19,8 @@ All LXC instances created using this repository come pre-installed with Midnight
 
 - **Bookstack LXC** [(View Source)](https://github.com/community-scripts/ProxmoxVE/blob/main/install/bookstack-install.sh)
   - NEW Script
+- **Vikunja LXC** [(View Source)](https://github.com/community-scripts/ProxmoxVE/blob/main/install/vikunja-install.sh)
```
Comment (havardthom) [inline CHANGELOG.md:22]: I think we should update changelog when we add the script to the website

**Comment by BramSuurdje** [comment]
script has been added to the website and will be added automatically after merge.

File: ct/vikunja.sh [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+if [[ ! -d /opt/vikunja ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+if (( $(df /boot | awk 'NR==2{gsub("%","",$5); print $5}') > 80 )); then
+  read -r -p "Warning: Storage is dangerously low, continue anyway? <y/N> " prompt
+  [[ ${prompt,,} =~ ^(y|yes)$ ]] || exit
+fi
+RELEASE=$(curl -s https://dl.vikunja.io/vikunja/ | grep -oP 'href="/vikunja/\K[0-9]+\.[0-9]+\.[0-9]+' | sort -V | tail -n 1)
+if [[ ! -f /opt/${APP}_version.txt ]] || [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]]; then
+  msg_info "Stopping ${APP}"
+  systemctl stop vikunja
+  msg_ok "Services ${APP}"
```
Comment (havardthom) [inline ct/vikunja.sh:68]:
```suggestion
  msg_ok "Stopped ${APP}"
```

File: install/vikunja-install.sh (hunk 9 above)
Comment (havardthom) [reply install/vikunja-install.sh:30]: @MickLesk was i wrong?

**Comment by havardthom** [comment]
You did not fix all comments before merging, why do we not have pull request ruleset enabled yet?

**Comment by MickLesk** [comment]
> You did not fix all comments before merging, why do we not have pull request ruleset enabled yet?

What is not fixed?

**Comment by havardthom** [comment]
The comment which is not resolved https://github.com/community-scripts/ProxmoxVE/pull/80#discussion_r1830084972

**Pull request merged**